connector/src/main/java/org/jboss/as/connector/subsystems/datasources/InstalledDriversListOperationHandler.java

```diff
@@ -70,10 +70,12 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
                             driverNode.get(DRIVER_XA_DATASOURCE_CLASS_NAME.getName());
 
                         } else {
+                            final int commaIndex = driver.getModuleName() != null ? driver.getModuleName().indexOf(":") : -1;
+                            final String moduleName = commaIndex != -1 ? driver.getModuleName().substring(0, commaIndex) : driver.getModuleName();
+                            final String moduleSlot = commaIndex != -1 ? driver.getModuleName().substring(commaIndex + 1) : "";
```

Comment:
I was thinking of consolidating this kind of thing in ModuleIdentifierUtil:

https://github.com/wildfly/wildfly-core/pull/6282/files#diff-867089bef7659adb3ead7e82d208a38aa95d3fda6df7476187f2aa272bf064d4R26-R41

So this could just be another function to pass in.

Reply:
It would be useful to have a utility method for this particular use case @bstansberry .